### DISCUSSION TITLE
Strange behavior on ESP32-S2

### DISCUSSION THREAD

**Discussion by Tellicious** [RavenSystem/esp-homekit-devices#2166] [MEPLHAA Scripts]
Hi José, I'm experiencing a very strange behavior on an ESP32-S2. The script itself should be pretty straight forward and when I tested with an ESP8266 it was working fine.

The two issues I'm having are:
1. When any button is pressed and released, regardless whether it's the on-board button (GPIO0) or the external one (GPIO21), the board enters setup mode as if the rising edge is not stopping the setup timer (both inputs are inverted) [Log 1](https://github.com/RavenSystem/esp-homekit-devices/files/13540031/Dec.03.203007.HAA.Log.log)
2. When the FM is started, even though it calculates correctly the value (1), the wildcard action is not called at all. I tried even setting `"v":0.5` but nothing changes [Log 2](https://github.com/RavenSystem/esp-homekit-devices/files/13540032/Dec.03.203242.HAA.Log.log)

Are you able to spot any mistake in my json? It's driving me crazy

I'm on HAA 12.9.1 on an ESP32-S2

```json
{
    "c": {
        "io": [
            [[1,2,3,4,5,6,7,8,9,10,11,13,14,16,17,18],0],
            [[15,34,33,36,35,38,37],2],
            [[0,21],6,1,1],
            [[12],6,0,1]
        ],
        "l": 15,
        "i": 0,
        "b":[[0,5],[21,5]],
        "o": 8,
        "10": {"m": [[1,0]]},
        "tt": [[10,2,0]],
        "n": "HAA-StereoControl",
        "ntp":"ntp1.inrim.it",
        "tz":"CET-1CEST-2,M3.5.0/2,M10.5.0/3"
    },
    "a": [
        {
            "0": {"m": [[4,-10001],[2,-20001]]},
            "1": {"m": [[2,-20000],[2,1]]},
            "b": [[0],[21]]
        },
        {
            "t": 2,
            "xa": 0,
            "0": {"r":[[33],[34,1,2]]},
            "1": {"r":[[34,1],[33,0,2]]},
            "f0": [[12,0]],
            "f1": [[12]]
        },
        {
            "h": 0,
            "g0": [[12,0]],
            "g1": [[12]],
            "s": 4
        },
        {
            "t": 80,
            "h": 0,
            "n": 5,
            "xa": 0,
            "ks": 0,
            "j": 2,
            "dt": [0,2,0,2,3,0],
            "y0":[{"v":1,"r":1,"0":{"m":[[2,0],[0,-10000]]}}]
        }
    ]
}
```

**Comment by RavenSystem** [comment]
1. It is caused by inverted buttons. With a pullup resistor, non pressed buttons will give 3.3V to GPIO, and GND when pressed. Inverted logic expects the opposite. Remove the inverted flag.
2. You set `"ks":0`, meaning that main and child flags are disabled. Child flag controls the execution of actions in Free Monitor service. In your case, set `"ks":2`. Sorry about this; I see it is not documented in the wiki.

**Reply by Tellicious** [reply]
Right! Don't know why, but after having configured the first 4 or 5 devices, I re-read the GPIO initialization documentation and understood it the other way around...and now I have like 8 devices running with the wrong configuration

Regarding the second point, is there any way to enable also child flags it via service notifications? -10001 seems to enable only the main flag

Thanks again!
